### ISSUE TITLE
c# selenium navigate().gotourl timing out when authentication box appears

### ISSUE THREAD

**Issue by CelDraTechSolutions** [SeleniumHQ/selenium#7714]
This issue has been raised several times but I'm yet to find a resolution or a way forward. Basically, since moving from Windows 7 to Windows 10 and IE 11, I'm encountering the following when executing Driver.Navigate().GoToUrl("URL HERE"):

"Timed out waiting for page to load."

   at OpenQA.Selenium.Remote.RemoteWebDriver.UnpackAndThrowOnError(Response errorResponse)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.set_Url(String value)
   at OpenQA.Selenium.Remote.RemoteNavigator.GoToUrl(String url)
   at TM10.PUB.StepDefinitions.PubSteps.NavigateToSite() in {LOCAL PATH TO SOLUTION FILE HERE}:line 37
   at TM10.PUB.StepDefinitions.PubSteps.GivenIAmOnThePublicationHomePage() in {LOCAL PATH TO SOLUTION FILE HERE}:line 31
   at lambda_method(Closure , IContextManager )

The following issues suggest that Selenium and IE no longer support automation of the security authentication dialogue box (login prompt where a username and password is entered) although the timeout is actually occurring within the Driver.Navigate().GoToUrl("URL HERE") command. We use AutoIt to execute against the login window but the script is failing before that.

https://github.com/SeleniumHQ/selenium/issues/6098
https://github.com/SeleniumHQ/selenium/issues/5932
https://github.com/SeleniumHQ/selenium/issues/5278

Basically, we need a proven way around this as a number of our sites which we have existing automated tests for require a user to login and having the facility to handle this is required.

For info, when we go back to using a Windows 7 machine with IE 10, the tests all work fine.

**Comment by pradeipp** [comment]
What worked for me is 
1. Using the 32-bit IEDriver instead of the 64-bit one
2. Adding a registry entry in the windows registry as suggessted -->[HERE](https://github.com/SeleniumHQ/selenium/wiki/InternetExplorerDriver/#user-content-required-configuration)<-- in the official docs
3. Handling the popup using simple Desktop UI Automation Code (I used **FlaUI**, a free open-source desktop UI automation tool)

**Comment by jimevans** [comment]
The “proven” way around authentication dialogs is to use a web proxy. There are tons of examples of how to do this on the web, including a [blog post series](https://jimevansmusic.blogspot.com/2019/05/handling-authentication-requests-with.html) on the topic I wrote myself using C#. The advantage of that approach is that it’s cross-browser, cross-platform, and will work in the RemoteWebDriver case as well as local execution.

**Comment by CelDraTechSolutions** [comment]
> The “proven” way around authentication dialogs is to use a web proxy. There are tons of examples of how to do this on the web, including a [blog post series](https://jimevansmusic.blogspot.com/2019/05/handling-authentication-requests-with.html) on the topic I wrote myself using C#. The advantage of that approach is that it’s cross-browser, cross-platform, and will work in the RemoteWebDriver case as well as local execution.

Thanks for taking the time to respond. Unfortunately, I've been struggling to get the proxy approach to work. I can debug through a test a couple of times before eventually I encounter proxy problems which prevent me from accessing any website. If I run the tests they just fail straight away due to proxy problems. My pc is on a network which has it's own proxy settings so I'm not sure if this is causing problems.

**Comment by CelDraTechSolutions** [comment]
> What worked for me is
> 
> 1. Using the 32-bit IEDriver instead of the 64-bit one
> 2. Adding a registry entry in the windows registry as suggessted -->[HERE](https://github.com/SeleniumHQ/selenium/wiki/InternetExplorerDriver/#user-content-required-configuration)<-- in the official docs. The 32 and 64 bit versions have slightly different entries, it's specified in the link. Keep an eye out for it.
> 3. Handling the popup using simple Desktop UI Automation Code (I used **FlaUI**, a free open-source desktop UI automation tool)

Thanks for the response. Can you confirm which version of the IEDriver you used, the version of FlaUI and code snippets please?
